platforms/common/src/lib.rs

```diff
@@ -4,3 +4,21 @@
 
 #[cfg(target_arch = "riscv32")]
 pub mod pmp_config;
+
+#[macro_export]
+macro_rules! read_volatile_at {
+    ($slice:expr, $index:expr) => {{
+        let ptr = unsafe { $slice.as_ptr().add($index) };
+        unsafe { core::ptr::read_volatile(ptr) }
+    }};
+}
+
+#[macro_export]
+macro_rules! read_volatile_slice {
+    ($slice:expr, $index:expr, $buf:expr) => {{
+        for (i, b) in $buf.iter_mut().enumerate() {
```

Comment:
I think at the very least we should add `assert!` to do the bounds check, since we are bypassing the normal slice bounds checks. Otherwise, this macro is still unsafe.